samza-core/src/main/java/org/apache/samza/runtime/RemoteApplicationRunner.java

```diff
@@ -106,11 +106,11 @@ public ApplicationStatus status(StreamApplication app) {
       }
 
       if (unsuccessfulFinish) {
-        return ApplicationStatus.UnsuccessfulFinish;
+        return ApplicationStatus.unsuccessfulFinish(null); // no exception for remote execution.
```

Comment:
Feels like this is a gap. Ideally, we should report an exception for the remote case too. Maybe, for example YarnJob should have the smarts to determine why an application failed using the Diagnostic Info